product_default_image/models/product_template.py

```diff
@@ -0,0 +1,317 @@
+            ('image_type', 'in', from_types),
+        ]
+
+        if add_domain:
+            domain += add_domain
+
+        return self.search(domain)
+
+    @api.multi
+    @api.onchange('categ_id')
+    def _onchange_categ_id(self):
+
+        company = self.env.user.company_id
+        target = company.product_image_target
+
+        if target in (NONE, GLOBAL):
+            return
+
+        for record in self:
+
+            if not record.auto_change_image:
+                continue
+
+            if record.image_type == CUSTOM:
+                continue
+
+            categ = record.categ_id
+
+            img_args = {
+                'to_type': CATEGORY,
+                'in_cache': True,
+            }
+            if not categ.image and target == GLOBAL_CATEGORY:
+                img_args['to_type'] = GLOBAL
+
+            record._change_template_image(**img_args)
+
+    @api.multi
+    def _change_template_image(self, to_type, to_img_bg=None, in_cache=False):
+        """ Change product template image to specified to_type's image
+            or to_img_bg.
+
+            Args:
+                to_type (str): Value specified allowed from image_type field
+                    which the image should be targeted to. Accepted values are:
+
+                    GLOBAL
+                        * product_image field defined in res.company
+
+                    CATEGORY
+                        * product's categ_id field's image
+
+                    NONE
+                        * Deletes the image
+
+                    CUSTOM
+                        * Set to the image you specify using to_img_bg.
+                        Note that targeting to custom will prevent the
+                        image from automatically being changed in the
+                        future until deleted or manually re-mapped to a
+                        default image.
+
+                to_img (base64): Can be used as an override if the template
+                    images should be changed to a specific image.
+                    Image is resized to image big value and scaled
+                    down to medium and small values.
+
+                    Note that to_type is required even if to_img_bg
+                    arg is supplied.
+
+                in_cache (bool): Set True if calling this method to change the
+                    images in cache (helpful when calling from an onchange
+                    method in product.template)
+
+            Returns:
+                ProductTemplate: Recordset if tmpl images have been changed.
+                None: if any of the conditions below are True.
+
+        """
+        if any([
+                to_type not in (NONE, GLOBAL, CATEGORY, CUSTOM),
+                to_type == GLOBAL_CATEGORY,
+                to_type == CUSTOM and not to_img_bg,
+                to_type == NONE and to_img_bg,
+                not isinstance(in_cache, bool)]):
+            return
```

Comment:
Yeah I was debating about that for awhile. I think perhaps it may be best to separate these into separate checks and raise errors accordingly.